### PR TITLE
feature: support take over all local driver volumes when execute d2p tool

### PR DESCRIPTION
**Pull request by HusterWan**
Signed-off-by: Michael Wan <zirenwan@gmail.com>

Before design d2p-migrator,  we think the local volumes is not a problem, we can clean them by using other tools. So we use `Bind` mode to take over those local volumes that making those local volumes cannot be deleted by `pouch` API. 

Those volumes have taken much disk space,  some even take all host disk space. So, we have to take over those local volumes now, and let them can be deleted by pouch volume delete API. 

Since the docker volume metadata information is not compacibility with pouch, So we must use the docker volume information to recreate pouch volume, there is what this PR done.

### CODE REVIEW

File: migrator/migrator.go [hunk 1]
```diff
@@ -157,7 +162,23 @@ func (p *PouchMigrator) PreMigrate(ctx context.Context, takeOverContainer bool)
...
-	containersDir := path.Join(p.pouchHomeDir, "containers")
+	// Get all volumes on host.
+	volumes, err := p.dockerd.VolumeList()
+	if err != nil {
+		return fmt.Errorf("failed to get volumes list: %v", err)
+	}
+	// check volume's driver, now only support local driver
+	for _, vol := range volumes {
+		if vol.Driver != "local" {
+			return fmt.Errorf("d2p only support local volume driver, now contains %s driver", vol.Driver)
```
Comment (allencloud) [inline migrator/migrator.go]: s/now/not ? 
Actually I cannot quite follow you.

File: migrator/migrator.go [hunk 2]
```diff
@@ -236,6 +326,23 @@ func (p *PouchMigrator) save2Disk(homeDir string, meta *pouch.PouchContainer) er
 	return nil
 }
 
+func (p *PouchMigrator) checkVolumeRefs(ctx context.Context, cid string, mounts []*pouchtypes.MountPoint, volumeRefs map[string]string) (map[string]string, error) {
```
Comment (allencloud) [inline migrator/migrator.go]: This file would update the data in the input parameter `volumeRefs`, does it make sense?

File: migrator/migrator.go (hunk 1 above)
Comment (HusterWan) [reply migrator/migrator.go]: Actually, what I mean is that we only support local driver now, but this host contains other volumes that created by other drivers.

Maybe my expression is to confusion 😆 i will updated it

File: migrator/migrator.go (hunk 2 above)
Comment (HusterWan) [reply migrator/migrator.go]: The parameter `volumeRefs` is to calculate  references by containers of a volume, so this function will change the data of `volumeRefs`.

If you have more reasonable advice, please let me know, thanks

**Comment by HusterWan** [comment]
Must wait after https://github.com/alibaba/pouch/pull/2392 merged, this PR can go on.

And i would like to invite @zhuangqh to review this PR

**Comment by zhuangqh** [comment]
LGTM

**Pull request merged**